### PR TITLE
[Feature] Analytics

### PR DESCRIPTION
**Pull request by Tysyatsky**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @NadiaKishchuk 
- [ ] @MementoMorj 
- [x] @MaksBrat 
- [ ] @Katerix 
- [x] @Kotusyk 
- [ ] @Kasterov 
- [x] @DanyilTerentiev 
- [ ] @Dobriyr 

## Summary of issue

Adding Google Analytics to the project with some custom events.

## Summary of change

Added current functionality for GA4.

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by MaksBrat — APPROVED** [review]
Great work!

**Review by DanyilTerentiev — APPROVED** [review]
Unfortunately not very into google analytics, but this code is SUUUPER dawg

File: public/index.html [hunk 1]
```diff
@@ -12,6 +12,14 @@
...
       Будуємо ком'юніті експертів."
   />
+  <!-- Google Tag Manager -->
+      <script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+      new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+      j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+      'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+      })(window,document,'script','dataLayer','GTM-5S25LMX');
+      </script>
+      <!-- End Google Tag Manager -->
```
Comment (Kotusyk) [inline public/index.html:22]: Does it need to be written in this code style way?

**Review by Kotusyk — COMMENTED** [review]
Great work, colleague!)

File: public/index.html (hunk 1 above)
Comment (Tysyatsky) [reply public/index.html:22]: yep, it's a copied from Google Analytics website to add tracking. So, it is not even mine)

**Pull request merged**